### PR TITLE
Add project rest apis

### LINKED ISSUE #164: Add project REST APIs
Need to provide the REST APIs to operate the project.

### PR DESCRIPTION
**Pull request by supereagle**
## Fixes #164

**Changes**

  - [x] New REST framework
  - [x] Project REST APIs

### CODE REVIEW

**Comment by supereagle** [comment]
@ddysher  @superxi911   Ready for review. PTAL, thanks:)

File: pkg/server/manager/project.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+// DeleteProject delete the project by name.
+func (m *projectManager) DeleteProject(projectName string) error {
+	project, err := m.dataStore.FindProjectByName(projectName)
+	if err != nil {
+		return err
+	}
+
+	// TODO (robin) Delete the pipelines belongs to this project.
```
Comment (superxi911) [inline pkg/server/manager/project.go]: This task need to be finished immediately

File: pkg/server/manager/pipeline.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+type PipelineManager interface {
+	// CRUD
+}
+
+type pipelineManager struct {
+}
+
+func NewPipelineManager(dataStore *store.DataStore) (PipelineManager, error) {
```
Comment (superxi911) [inline pkg/server/manager/pipeline.go]: comments

File: pkg/server/manager/project.go (hunk 1 above)
Comment (supereagle) [reply pkg/server/manager/project.go]: Depends on the implementation of pipeline, will finish this in next PR for pipeline.

File: pkg/server/manager/pipeline.go (hunk 2 above)
Comment (supereagle) [reply pkg/server/manager/pipeline.go]: Just a framework for pipeline, will add comments with implementation of pipeline in next PR.

File: pkg/server/manager/project.go (hunk 1 above)
Comment (superxi911) [reply pkg/server/manager/project.go]: 👌

**Pull request merged**